CMakeLists.txt

```diff
@@ -60,6 +63,17 @@ if(BUILD_TESTS)
     endif()
 endif()
 
-# Install commands
-install(TARGETS ale LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
-install(DIRECTORY ${ALE_INCLUDE_DIRS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+# Generate the package config
+configure_file(cmake/config.cmake.in
+               ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config.cmake
+               @ONLY)
+
+# Install the package config
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config.cmake
+              DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
+
+# Install the library
+install(TARGETS ale
+        EXPORT ale
```

Comment:
This and the previous config exports ALE as ale::ale. Read some docs, it looks like we always want to use a namespace scoping because it give better error checking when used downstream.